### PR TITLE
Docker Action: checkout v4

### PR DESCRIPTION
**Pull request by liyishuai**
> Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/upload-artifact@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.

### CODE REVIEW

**Review by Zimmi48 — APPROVED** [review]
Thanks! If this is tested, feel free to merge.

There are a number of other places where we should do a similar update BTW (e.g., the Nix template, the Coq Nix Toolbox, etc.).

**Comment by liyishuai** [comment]
@Zimmi48 are there pending tasks before merging this PR? The merge button is gray to me.

**Comment by Zimmi48** [comment]
Sorry, I missed your previous message!
I'm surprised that you cannot merge. Wasn't it the case before? Did we change anything in the configuration that could have had this effect @palmskog?
Anyway, merging this now.

**Pull request merged**
